frontend/src/metabase/sharing/components/SharingSidebar.jsx

```diff
@@ -496,9 +502,25 @@ class SharingSidebar extends React.Component {
     this.setState({ editingMode: "unknown" });
   };
 
+  // Because you can navigate down the sidebar, we need to wrap
+  // onCancel from props and either call that or reset back a screen
+  onCancel = () => {
+    const { onCancel } = this.props;
+    console.log("do we get here?", this.state.returnMode);
+    if (this.state.returnMode && this.state.editingMode !== "unknown") {
```

Comment:
I'm fine with this as defensive programming, but am curious about whether we'd ever be in a state where `returnMode` was set and `editingMode` _was_ `"unknown"`